src/components/Swap/Toolbar/index.tsx

```diff
@@ -61,7 +61,7 @@ function CaptionRow() {
       default:
     }
 
-    if (state === TradeState.LOADING) {
+    if (state === TradeState.LOADING && trade === undefined) {
```

Comment:
Can this be made '&& !trade' to better mirror the next check? This way if types change we'd know all cases are still covered here.